transport/response.go

```diff
@@ -24,5 +24,6 @@ import "io"
 
 // Response is the low level response representation.
 type Response struct {
-	Body io.ReadCloser
+	Headers map[string]string
+	Body    io.ReadCloser
```

Comment:
I'm ok with `transport.Properties` for now, can't think of anything better. Let's just make sure we have consistency between our lang impls.